thrift/interfaces.go

```diff
@@ -44,3 +44,44 @@ type TChanServer interface {
 	// Methods returns the method names handled by this server.
 	Methods() []string
 }
+
+// InterceptorRunnerRegistrar is a target for the registration of an
+// interceptor runner.
+type InterceptorRunnerRegistrar interface {
+	// RegisterInterceptorRunner registers the interceptor runner.
+	RegisterInterceptorRunner(InterceptorRunner)
+}
+
+// InterceptorPostRunner is a callback that should be exectuted after
+// a tchannel request has been handled.
+type InterceptorPostRunner func(response athrift.TStruct, err error) error
+
+// InterceptorRunner handles the excecution of a collection of
+// Interceptor Pre functions and returns a Interceptor post runner
+// that handles the execution of the Post methods of those same
+// Interceptors.
+type InterceptorRunner interface {
+	// RunPre will run the pre interceptors in order, and return
+	// a function for the post interceptors to run.
+	RunPre(ctx Context, method string, args athrift.TStruct) (
```

Comment:
I try to keep comments < 100 chars. If a line is like > 120, then i'll break it up at a place that seems logical

no hard rule though